core/eolearn/core/utils/testing.py

```diff
@@ -80,3 +83,15 @@ def _get_feature_shape(
 ) -> Tuple[int, ...]:
     time, height, width, depth = len(timestamps), *config.raster_shape, rng.integers(*config.depth_range)
     return (time, height, width, depth) if ftype.is_temporal() else (height, width, depth)
+
+
+def assert_feature_match(tested_feature: Any, expected_feature: Any) -> None:
+    if type(tested_feature) is not type(expected_feature):
+        assert type(tested_feature) is type(expected_feature)
```

Comment:
this is really weird... why have the assert in a conditional?
also directly comparing types is problematic for classes and inheritance

Reply:
> why have the assert in a conditional? 
you cant simply do `assert False`

probably the best solution is to remove this two lines, because if type are not the same the test fill fail any how ?

Reply:
you actually can do `assert False` :)
you can also have just `assert type(tested_feature) is type(expected_feature)`. But that is again vulnerable to subclasses, so something like `assert isinstance(a, type(b)) or isinstance(b, type(a))` would be needed.

If you remove them, then i'd call `isinstance` on both elements, `assert_arrays_equal` is not very picky about the type of input (but we want to be)

Reply:
> you actually can do assert False :)
you can but flake8 dose not relay like it

> If you remove them, then i'd call isinstance on both elements, assert_arrays_equal is not very picky about the type of input (but we want to be)
kinda like this option :)

Reply:
done